Sources/NIOTransportServices/StateManagedChannel.swift

```diff
@@ -200,15 +200,20 @@ extension StateManagedChannel {
 
             self.doClose0(error: error)
 
-            if case .active = oldState {
+            switch oldState {
+            case .active:
                 self.pipeline.fireChannelInactive()
+                fallthrough
```

Comment:
No, it's fine. The gating state transition is in `try self.state.becomeInactive()` several lines above. That line can only be run once, as it will transition all states to `.inactive` unless the state already is `.inactive`, in which case it will throw. As a result, we can only execute the body of `close0` once. Additionally, all other methods will check the state based on that state enum, rather than by asking if we're registered.

Reply:
cool :)